femtools/Detector_Tools.F90

```diff
@@ -335,14 +426,26 @@ subroutine delete_all_detectors(detector_list)
 
   end subroutine delete_all_detectors
 
+  subroutine delete_all_temp_detectors(detector_list)
```

Comment:
Again I wouldn't call it `delete_all_temp_detectors`as it does not delete or deallocate detectors.

I would just merge this with `temp_deallocate` above (and call it `temp_list_deallocate`). Strictly speaking it doesn't do any deallocation, it just wipes the temp links. Even that nullifying of the temp links in principle isn't necessary, as they'll simply be overwritten the next time a detector is inserted into a temp list. So in principle the user shouldn't have to do anything when it's done with the temp list, but it is probably neater to do this clean up (and allows to later change the implementation where some deallocation might be necessary).